apps/yapms/src/lib/components/modals/sharemodal/ShareModal.svelte

```diff
@@ -3,18 +3,64 @@
 	import ArrowDownTray from "$lib/icons/ArrowDownTray.svelte";
 	import ArrowUpTray from "$lib/icons/ArrowUpTray.svelte";
   import { ShareModalStore } from "$lib/stores/Modals";
-	import saveToJson from "$lib/utils/saveToJson";
-	import loadFromJson from "$lib/utils/loadFromJson";
+	import { downloadJson, generateJson } from "$lib/utils/saveMap";
+	import Link from "$lib/icons/Link.svelte";
+	import { loadFromFile } from "$lib/utils/loadMap";
+	import ExclamationCircle from "$lib/icons/ExclamationCircle.svelte";
+  import { page } from "$app/stores";
+	import CheckCircle from "$lib/icons/CheckCircle.svelte";
 
   let files: FileList;
 
+  let fetchingLinkID = false;
+  let copiedLinkID = false;
+  let linkID: string | null = null;
+
   function load() {
     if (files && files.length > 0) {
-      loadFromJson(files);
+      loadFromFile(files);
       ShareModalStore.set({ ...$ShareModalStore, open: false });
     }
   }
 
+  async function generateLink() {
+    const blob = new Blob(
+      [JSON.stringify(generateJson())],
+      { type: "application/json" }
+    );
+
+    const formData = new FormData();
+    formData.append("data", blob, "data.json");
+
+    const url = new URL("http://localhost:8080/api/yapms.com/map");
```

Comment:
Probably want to change localhost:8080 to an environment variable

Reply:
Fixed